core/templates/pages/exploration-player-page/services/question-player-engine.service.spec.ts

```diff
@@ -766,15 +803,13 @@ describe('Question player engine service ', () => {
       questionPlayerEngineService.init(
         [singleQuestionBackendDict], successHandler, failHandler);
 
-      let nextQuestion = null;
+      let createNewCardSpy = spyOn(
+        stateCardObjectFactory, 'createNewCard').and.returnValue(null);
 
       questionPlayerEngineService.submitAnswer(
         answer, interactionRulesService, successCallback);
 
-      expect(successCallback).toHaveBeenCalledWith(
-        nextQuestion, true, '', undefined, null,
-        null, false, 'misconceptionId',
-        null, null, true, 'focusLabel');
+      expect(createNewCardSpy).not.toHaveBeenCalled();
```

Comment:
Same note as above.

You might be better served having both of these in the same test. Have 3 questions perhaps. Show that the behaviour for the first two is different from the last one. That serves as the pre-check and the post-check (currently you have two tests each with only one check, which is weaker).

Reply:
done!